atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTable.java

```diff
@@ -243,26 +250,80 @@ private void executeIgnoringError(String sql, String errorToIgnore) {
     }
 
     @Override
-    public void compactInternally() {
+    public void compactInternally(boolean inSafeHours) {
+        final String compactionFailureTemplate = "Tried to clean up {} bloat after a sweep operation,"
+                + " but underlying Oracle database or configuration does not support this {} feature online. "
+                + " Since this can't be automated in your configuration,"
+                + " good practice would be do to occasional offline manual maintenance of rebuilding"
+                + " IOT tables to compensate for bloat. You can contact Palantir Support if you'd"
+                + " like more information. Underlying error was: {}";
+
         if (config.enableOracleEnterpriseFeatures()) {
             try {
-                conns.get().executeUnregisteredQuery(
+                getCompactionConnection().executeUnregisteredQuery(
                         "ALTER TABLE " + oracleTableNameGetter.getInternalShortTableName(conns, tableRef)
                                 + " MOVE ONLINE");
             } catch (PalantirSqlException e) {
-                log.error("Tried to clean up {} bloat after a sweep operation,"
-                        + " but underlying Oracle database or configuration does not support this"
-                        + " (Enterprise Edition that requires this user to be able to perform DDL operations)"
-                        + " feature online. Since this can't be automated in your configuration,"
-                        + " good practice would be do to occasional offline manual maintenance of rebuilding"
-                        + " IOT tables to compensate for bloat. You can contact Palantir Support if you'd"
-                        + " like more information. Underlying error was: {}",
-                        tableRef.toString(), e.getMessage());
+                log.error(compactionFailureTemplate,
+                        tableRef,
+                        "(Enterprise Edition that requires this user to be able to perform DDL operations)."
+                                + " Please change the `enableOracleEnterpriseFeatures` config to false.",
+                        e.getMessage());
+            } catch (TableMappingNotFoundException e) {
+                throw Throwables.propagate(e);
+            }
+        } else if (config.enableShrinkOnOracleStandardEdition()) {
+            Stopwatch timer = Stopwatch.createStarted();
+            try {
+                if (inSafeHours) {
```

Comment:
Maybe too late but 'safe hours' could be misinterpreted.  Does it mean "safe to perform locking DB operations" or "keep these hours safe as users are using the system".  Maybe 'maintenance hours', or 'table move hours'.  I don't feel strongly though so if the messaging is clear where this is configured then ignore this.

Reply:
Right! I think `maintenanceHours` is substantially clearer - will rename to this.